ecl/hql/hqlmeta.cpp

```diff
@@ -1573,8 +1573,9 @@ IHqlExpression * ensureSorted(IHqlExpression * dataset, IHqlExpression * order,
         }
     }
 
-    IHqlExpression * attr = isLocal ? createLocalAttribute() : (isGrouped(dataset) && ignoreGrouping) ? createAttribute(globalAtom) : NULL;
-    return createDatasetF(no_sort, LINK(dataset), LINK(order), attr, NULL);
+    IHqlExpression * attr1 = isLocal ? createLocalAttribute() : (isGrouped(dataset) && ignoreGrouping) ? createAttribute(globalAtom) : NULL;
+    IHqlExpression * attr2 = requestSpilling ? createAttribute(internalAtom) : NULL;
+    return createDatasetF(no_sort, LINK(dataset), LINK(order), attr1 ? attr1 : attr2, attr1 ? attr2 : NULL, NULL);
```

Comment:
Not sure I like the conditional attributes.  createDataset has the (slightly unusual) semantic of expanding out no_comma.  So cleaner would be:

return createDatasetF(no_sort, LINK(dataset), options, createComma(attr1, attr2), NULL):